### PR TITLE
Supports broker-to-client topic aliases

### PR DESCRIPTION
**Pull request by zhanghongtong**
Supports broker-to-client topic aliases

### CODE REVIEW

**Comment by zhanghongtong** [comment]
[Topic Alias](https://trello.com/c/Q9EzWOFK/100-topic-alias)

**Comment by tigercl** [comment]
[Read new configuration items when plugin is loaded](https://trello.com/c/yhW1Q9Ju/108-read-new-configuration-items-when-plugin-is-loaded)

File: src/emqx_channel.erl [hunk 1]
```diff
@@ -658,45 +661,49 @@ return_connack(AckPacket, Channel) ->
...
 do_deliver([Publish], Channel) ->
     do_deliver(Publish, Channel);
 
 do_deliver(Publishes, Channel) when is_list(Publishes) ->
-    lists:reverse(
-      lists:foldl(
-        fun(Publish, Acc) ->
-                lists:append(do_deliver(Publish, Channel), Acc)
-        end, [], Publishes)).
+    do_deliver(Publishes, [], Channel).
```
Comment (HJianBo) [inline src/emqx_channel.erl]: The previous style may cleaner than it

File: src/emqx_channel.erl [hunk 2]
```diff
@@ -1085,12 +1092,42 @@ process_alias(#mqtt_packet{
...
+%% Packing Topic Alias
+
+packing_alias(Packet = #mqtt_packet{
+                          variable = #mqtt_packet_publish{topic_name = Topic} = Publish
+                         },
+              Channel = #channel{topic_aliases = TopicAliases, alias_maximum=Limits}) ->
+    case find_alias(outbound, Topic, TopicAliases) of
+        {ok, AliasId} -> 
+            NPublish = Publish#mqtt_packet_publish{topic_name = <<>>,
+                                                    properties = #{'Topic-Alias' => AliasId}
```
Comment (HJianBo) [inline src/emqx_channel.erl]: It should follow the previous line indent

File: src/emqx_channel.erl [hunk 3]
```diff
@@ -1085,12 +1092,42 @@ process_alias(#mqtt_packet{
...
+        error ->
+            #{outbound := Aliases} = TopicAliases,
+            AliasId = maps:size(Aliases) + 1,
+            case (Limits =:= undefined) orelse
+                    (AliasId =< maps:get(outbound, Limits, 0)) of
+                true ->
+                    NTopicAliases = save_alias(outbound, AliasId, Topic, TopicAliases),
+                    NChannel = Channel#channel{topic_aliases = NTopicAliases},
+                    NPublish = Publish#mqtt_packet_publish{topic_name = Topic,
+                                                    properties = #{'Topic-Alias' => AliasId}
```
Comment (HJianBo) [inline src/emqx_channel.erl]: It should follow the previous line indent

**Pull request merged**